### DISCUSSION TITLE
Finding the best way to use the parent component's state in a tag or node

### DISCUSSION THREAD

**Discussion by khash** [markdoc/markdoc#183] [Q&A] [answered]
I am using MarkDoc with NextJS.

In my `Layout.jsx` I have a state, called `currentFramework` and I need to pass this to a custom tag. I have a component that accepts this as a property and shows or hides its children:

```js
<PerFramework currentFramework={currentFramework} includes={['rails', 'react']}>
   <span>This is for Rails and React only</span>
</PerFramework>
```

Now I want to make this component available as a tag, with something like this:

```
{% per-framework includes=["rails", "react"] currentFramework=??? %}
    This is for Rails and Node Only
{% /per-framework %}
```

Ideally, I'd like to do this without the need for the author to include `currentFramework` on every tag, and somehow pass the state into the render process during Transform.

**Comment by khash** [comment, marked as answer]
I managed to solve this problem by using React Context, but not sure if that's the best approach:

- Created a React Context and wrapped the layout in it.
- Used the context in the PerFramework component to determine show and hide
- Set the context at the layout

**Reply by mfix-stripe** [reply]
@khash came here to say this problem is probably best suited for React Context, as far as I can tell. You can also set the value in somewhere like the frontmatter and feed it through to the `per-framework` `transform` function, but I think your solution is best.
